File: solved.ac/javascript/13164.js
//https://hyunjinee.tistory.com/32
const inputs = require("fs").readFileSync("./dev/stdin").toString().split("\n")
let [n, k] = inputs[0].split(" ").map(Number)
let heights = inputs[1].split(" ").map(Number)
let diff = []
for (let i = 1; i < n; i++) {
  diff.push(heights[i] - heights[i - 1])
}
diff.sort((a, b) => b - a)
diff = diff.slice(k - 1)
const ans = diff.reduce((acc, cur) => acc + cur, 0)
console.log(ans)

File: solved.ac/python/13164.py
# https://hyunjinee.tistory.com/32
import sys
input = sys.stdin.readline
n,k = map(int,input().split())
heights = list(map(int,input().split()))
costs = []
for i in range(n-1):
  costs.append(heights[i+1]-heights[i])
costs.sort(reverse=True)
print(sum(costs[k-1:]))